src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -102,22 +106,190 @@ export default class NativeAdapter implements IMediaSourceAdapter {
   }
 
   /**
+   * @constructor
+   * @param {HTMLVideoElement} videoElement - The video element which bind to NativeAdapter
+   * @param {string} source - The source URL
+   * @param {Object} config - The media source adapter configuration
+   */
+  constructor(videoElement: HTMLVideoElement, source: Object, config: Object) {
+    this._config = config;
+    this._videoElement = videoElement;
+    this._source = source.url;
+    this._eventManager = new EventManager();
+  }
+
+  /**
    * Load the video source
    * @function load
-   * @returns {void}
+   * @returns {Promise<Object>} - The loaded data
    */
-  load(): void {
-    NativeAdapter._logger.debug('load');
-    this._videoElement.load();
+  load(): Promise<Object> {
+    if (!this._loadPromise) {
+      this._loadPromise = new Promise((resolve, reject) => {
+        this._eventManager.listen(this._videoElement, PlayerEvents.LOADED_METADATA, () => {
+          this._eventManager.unlisten(this._videoElement, PlayerEvents.LOADED_METADATA);
+          let data = {tracks: this._getParsedTracks()};
+          NativeAdapter._logger.debug('load');
```

Comment:
@yairans please be more accurate with log output, one word is not beneficial for logs clarity